homework3/task01/project_cache/task1cache.py

```diff
@@ -0,0 +1,53 @@
+"""
+Task.
+
+In previous homework 2 task 4, you wrote a cache function that remembers other function output value.
+Modify it to be a parametrized decorator, so that the following code:
+
+@cache(times=3)
+def some_function():
+    pass
+Would give out cached value up to times number only. Example:
+
+@cache(times=2)
+def f():
+    return input('? ')   # careful with input() in python2, use raw_input() instead
+
+f()
+? 1
+'1'
+f()     # will remember previous value
+'1'
+f()     # but use it up to two times only
+'1'
+f()
+? 2
+'2'
+"""
+from typing import Any, Callable
+
+
+def cache(times: int) -> Callable:
+    """
+    Remember other function output value. Give out cached value up to times number.
+
+    :times: quantity of returned cached values
+    """
+
+    def wrapper(func: Callable) -> Callable:
+        cache_data = {}
+        if times < 1:
+            raise ValueError("Times value should be more than zero")
+
+        def timer(*args: Any) -> Any:
```

Comment:
I'd say this name usually used for different purposes, so maybe name it like `cache_with_counter`